spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraTableRowReader.scala

```diff
@@ -0,0 +1,175 @@
+package com.datastax.spark.connector.rdd
+
+import java.io.IOException
+
+import com.datastax.driver.core.{ProtocolVersion, Session}
+import com.datastax.spark.connector._
+import com.datastax.spark.connector.cql.{CassandraConnector, Schema}
+import com.datastax.spark.connector.rdd.reader._
+
+import scala.reflect.ClassTag
+
+/**
+ * Used to get a RowReader of type [R] for transforming the rows of a particular Cassandra table into
+ * scala objects. Performs necessary checking of the schema and output class to make sure they are
+ * compatible.
+ * @see [[CassandraTableScanRDD]]
+ * @see [[CassandraJoinRDD]]
+ */
+trait CassandraTableRowReader[R] {
```

Comment:
Hmm, this is still confusing, because the name suggests this object is a RowReader. While in fact it only provides a RowReader to the RDD implementation. Maybe call it `CassandraTableRowReaderProvider`?